packages/openapi-typescript/test/transform/schema-object/composition.test.ts

```diff
@@ -258,20 +258,12 @@ describe("composition", () => {
       "discriminator > oneOf",
       {
         given: {
-          type: "object",
```

Comment:
@drwpow I was unsure why there is already and object here for the test of a oneOf discriminator, so I removed it. Maybe the implicit allOf merging here should be in a separate unit test?

Reply:
Yes this actually should be its own test, you’re right, but it’s important we keep this “wrong” testcase. We still get so many bug reports and code examples of people combining `type: "object"` with its own properties + `oneOf` (even though in the docs it suggests not doing that). You’re right—it shouldn’t be the _ideal_ case, but we should keep it as _a_ test case so we can just be sure that it generates _something_ (logical collisions are the responsibility of the schema author, but openapi-typescript should still attempt to generate something that reflects what they’ve authored)